### PR TITLE
MLPAB-2054: Block manual navigation to restricted pages

### PR DESCRIPTION
**Pull request by acsauk**
# Purpose

Fixes MLPAB-2054

### CODE REVIEW

File: docker/mock-lpa-store/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,10 @@
+ARG IMPOSTER_VERSION=$IMPOSTER_VERSION
```
Comment (acsauk) [inline docker/mock-lpa-store/Dockerfile:1]: Looks weird but we need ARG here to make it available to the second `FROM`

File: internal/page/attorney/register.go [hunk 2]
```diff
@@ -186,6 +186,10 @@ func makeAttorneyHandle(mux *http.ServeMux, store SessionStore, errorHandler pag
 				return
 			}
 
+			if !page.AttorneyCanGoTo(attorney, r.URL.String()) {
+				http.Redirect(w, r, appData.Lang.URL(page.Paths.Attorney.TaskList.Format(attorney.LpaID)), http.StatusFound)
```
Comment (hawx) [inline internal/page/attorney/register.go:190]: `page.Paths.Attorney.TaskList.Redirect(w, r, appData, attorney.LpaID)` is neater for these, also you'll need to `return` after them or it runs through and produces an error

File: web/template/layout/address-summary-row.gohtml [hunk 3]
```diff
@@ -10,7 +10,7 @@
                 </a>
             {{ end }}
         </dd>
-        {{ if and .Address .CanChange }}
+        {{ if and ( not (eq .Address.Line1 "")) .CanChange }}
```
Comment (hawx) [inline web/template/layout/address-summary-row.gohtml:13]: you can do `ne ...` instead of `not (eq ...)`

Comment (acsauk) [reply web/template/layout/address-summary-row.gohtml:13]: Much neater

File: internal/page/attorney/register.go [hunk 4]
```diff
@@ -187,7 +187,11 @@ func makeAttorneyHandle(mux *http.ServeMux, store SessionStore, errorHandler pag
 			}
 
 			if !page.AttorneyCanGoTo(attorney, r.URL.String()) {
-				http.Redirect(w, r, appData.Lang.URL(page.Paths.Attorney.TaskList.Format(attorney.LpaID)), http.StatusFound)
+				if err := page.Paths.Attorney.TaskList.Redirect(w, r, appData, attorney.LpaID); err != nil {
```
[the remark this answers is not in the document]
Comment (acsauk) [reply internal/page/attorney/register.go]: Ah gotcha 👍

**Review by hawx — APPROVED** [review]

**Pull request merged**